src/components/Header/Header.tsx

```diff
@@ -65,7 +65,22 @@ class Header extends UIComponent<any, any> {
       )
     }
 
-    const descriptionElement = HeaderDescription.create(description, { generateKey: false })
+    const descriptionProps =
+      typeof descriptionContentOrProps === 'string'
+        ? { content: descriptionContentOrProps }
+        : descriptionContentOrProps
+
+    const descriptionElement = HeaderDescription.create(
+      {
+        ...descriptionProps,
+        variables: {
+          ...(v.descriptionColor && { color: v.descriptionColor }),
+        },
+      },
+      {
+        generateKey: false,
+      },
+    )
```

Comment:
This can be simplified with the following:
```
    const descriptionElement = HeaderDescription.create(
      descriptionContentOrProps,
      {
        defaultProps: {
          variables: {
            ...(v.descriptionColor && { color: v.descriptionColor }),
          },
        },
        generateKey: false,
      },
    )
```

Reply:
thing that I am worrying about is that this is semantically not the same as providing these custom variables as `variables` property - here, with the suggested approach, those are passed as `defaultProps`. Shouldn't we instead consider `overrideProps` from this config object at least?